### PR TITLE
Cmos support

### PR DESCRIPTION
**Pull request by omarandlorraine**
I figure we'll want CMOS support here as well.

I'm leaving this as a draft pull request until I've implemented all the new instructions and other differences

### CODE REVIEW

**Comment by omarandlorraine** [comment]
Oops, you see a lot of these commits come from the NMOS indirect jump PR which is #98; this pull request will depend on that one.

File: src/instruction.rs [hunk 1]
```diff
@@ -507,7 +620,32 @@ impl crate::Variant for Cmos6502 {
     fn decode(opcode: u8) -> Option<(Instruction, AddressingMode)> {
         // TODO: We obviously need to add the other CMOS instructions here.
```
Comment (mre) [inline src/instruction.rs:621]: Is this comment still relevant? If so, should we list the missing instructions for future reference here?

**Review by mre — APPROVED** [review]

**Comment by mre** [comment]
Looks good to me and can be merged. Good job!

File: src/instruction.rs (hunk 1 above)
Comment (omarandlorraine) [reply src/instruction.rs:621]: As far as I can tell the missing instructions are to do with interrupt handling, and so depend on #64 for me to do anything useful here

**Comment by omarandlorraine** [comment]
> Looks good to me and can be merged. Good job!

Thanks!

**Pull request merged**